hw4.js

```diff
@@ -0,0 +1,475 @@
+let obj2 = { age: 11 };
+const mergeObjects = (objects) => {
+    const result = { ...obj1, ...obj2 };
+    return result;
+};
+
+/**
+ * Exercise 7
+ *
+ * Write a function that returns the smallest value of an array
+ * [5,200,-5,41] => -5
+ *
+ * Use: operator ... and Math.min
+ */
+let arr7 = [5, 200, -5, 41];
+const getSmallestValue2 = (numbers) => {
+    let result = Math.min(...arr7);
+    return result;
+};
+
+/**
+ * Exercise 8
+ *
+ * Write a function that returns odd array values.
+ * [77,2,30,51] => [77,51]
+ *
+ * Use: reduce
+ */
+let arr8 = [77, 2, 30, 51];
+const getOddValues2 = (numbers) => {
+    let reducer = (a, b) => (b % 2 === 0 ? a : a.concat(b));
+    let result = arr8.reduce(reducer, []);
+    return result;
+};
+
+/**
+ * Exercise 9
+ *
+ * Write a function that accepts data from the basket as input in the following form:
+ *
+ *[
+ * {price: 10, count: 2},
+ * {price: 100, count: 1},
+ * {price: 2, count: 5},
+ * {price: 15, count: 6},
+ *]
+ * where price is the price of the item and count is the quantity.
+ *
+ * The function should return the total price for this order.
+ *
+ * Use: reduce
+ */
+let arr9 = [
+    { price: 10, count: 2 },
+    { price: 100, count: 1 },
+    { price: 2, count: 5 },
+    { price: 15, count: 6 },
+];
+const calculateTotal = (products) => {
+    let reducer = (a, b) => ({ price: a.price + b.price });
+    let result = arr9.reduce(reducer, { price: 0 });
+    return result;
+};
+
+/**
+ * Exercise 10
+ *
+ * Implement a function that has an array of numbers as input and an array of unique values as output
+ * [1, 2, 2, 4, 5, 5] => [1, 2, 4, 5]
+ *
+ * Use: reduce and indexOf
+ */
+let arr10 = [1, 1, 2, 3, 4, 4];
+const getUniqueValues = (numbers) => {
+    return arr10.reduce((uniqueArray, number) => {
+        if (uniqueArray.indexOf(number) === -1) {
+            uniqueArray.push(number);
+        }
+        return uniqueArray;
+    }, []);
+};
+
+/**
+ * Exercise 11
+ *
+ * Implement a function whose input is a numeric code of an error, the output is a string with a message
+ * 500 => Server Error
+ * 401 => Authorization failed
+ * 402 => Server Error
+ * 403 => Access denied
+ * 404 => Not found
+ *
+ * Use: switch case or object like a map structure
+ */
+let errorCode = 500;
+const getErrorMessage = (code) => {
+    let result = "";
+    switch (errorCode) {
+        case 500:
+        case 402:
+            result = "Server Error";
+            break;
+        case 401:
+            result = "Authorization failed";
+            break;
+        case 403:
+            result = "Access denied";
+            break;
+        case 404:
+            result = "Not found";
+            break;
+    }
+    return result;
+};
+
+/**
+ * Exercise 12
+ *
+ * Write a function that returns the 2 smallest values of an array
+ * [4,3,2,1] => [1,2]
+ *
+ * Use: .sort()
+ */
+let arr12 = [4, 3, 2, 1];
+const get2SmallestValues = (numbers) => {
+    let arrAfterSort = arr12.sort();
```

Comment:
use custom callback to sort items, sorting by default doesn't work as good as expected

Reply:
get2SmallestValues([4, 3, -1, -3, -7, 2, 1])

expected result: [-7, -3]
actual result: [-1, -3]